src/cite.cpp

```diff
@@ -152,12 +153,48 @@ void CitationManager::generatePage()
             if (j!=-1 && k!=-1)
             {
               QCString label = line.mid((uint)(j+1),(uint)(k-j-1));
-              if (p->entries.find(label.data())==p->entries.end()) // not found yet
+              // check if the reference with the cross reference is used
+              // insert cross refererence when cross reference has not yet been added.
+              if ((p->entries.find(label1.data())!=p->entries.end()) &&
+                  (p->entries.find(label.data())==p->entries.end())) // not found yet
               {
                 insert(label);
               }
             }
           }
+          else if (line.stripWhiteSpace().startsWith("@"))
+          {
+            // assumption entry like: "@book { name," or "@book { name" (spaces optional)
+            int j=line.find("{",0);
+            // when no {, go hunting for it
+            while (j==-1 && (s=input.find('\n',pos))!=-1)
+            {
+              line = input.mid((uint)pos,(uint)(s-pos));
+              j=line.find("{",0);
+              pos=s+1;
+            }
+            // search for the name
```

Comment:
Is it guaranteed that a "{" will always be found? otherwise j will be -1 and this will be used below in find...

tip: to find a character starting at position 0, one can do `line.find('{')` instead of `line.find("{",0)` which is shorter and more efficient.

A comment with a typical input example and what is ultimately assigned to `label1` would help to understand the code... if possible also use a more descriptive variable name.

Reply:
- tip about `find`, implemented
- replaced `label..` with more descriptive names
- found some regressions, especially when no `citeName` is present, this only occurs at the end of a bib fie (and actually the bib  is not correct). At the end of the bib file one has to add something like:
  - `@book ,`
  - `@book { ,`